test/cluster/test_tablets_lwt.py

```diff
@@ -638,5 +637,68 @@ async def test_error_message_for_timeout_due_to_uncertainty(manager: ManagerClie
         logger.info("Trigger paxos_accept_proposal_wait")
         await manager.api.message_injection(servers[1].ip_addr, "paxos_accept_proposal_wait")
 
-        with pytest.raises(ReadTimeout, match="write timeout due to uncertainty(.*)injected_error_before_save_proposal"):
+        with pytest.raises(WriteTimeout, match="write timeout due to uncertainty(.*)injected_error_before_save_proposal"):
             await lwt_task
+
+
+@pytest.mark.asyncio
+@skip_mode('debug', 'dev is enought')
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_no_uncertainty_for_reads(manager: ManagerClient):
+    # This test verifies that LWT reads do not produce 'uncertainty' timeouts.
+    #
+    # The scenario is similar to the write-uncertainty test:
+    # 1. The cluster has three nodes. The first node is configured to fail on accept,
+    #    and the second node is configured to block until explicitly signaled by the test.
+    # 2. Run a read LWT on the first node and wait until the coordinator processes an accept error
+    #    from the first node.
+    # 3. Run another LWT, which should invalidate all promises from the first one.
+    # 4. Signal the second node to proceed with accept; it must return a reject.
+    # 5. At this point, the coordinator is uncertain about the outcome of the first LWT. Any subsequent LWT
+    #    might either complete its Paxos round (if a quorum of promises observed the accept) or
+    #    overwrite it (if a quorum did not). Since the first LWT is read-only, the coordinator retries it
+    #    transparently and returns the effects of the second LWT to the client.
+
+    logger.info("Bootstrapping cluster")
+    cmdline = [
+        '--logger-log-level', 'paxos=trace'
+    ]
+    servers = await manager.servers_add(3, cmdline=cmdline, auto_rack_dc="mydc")
```

Comment:
Q (unrelated): can we recruit an existing 3-node cluster rather than create a new one?

Reply:
I don't think so, everybody is just creating a new cluster nowadays. It's not easy to reuse them because a test can leave the cluster in some undefined state. Maybe @xtrey can comment.